gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -625,14 +670,17 @@ private int peekKeyword() throws IOException {
       return PEEKED_NONE;
     }
 
-    // Confirm that chars [1..length) match the keyword.
+    // Upper cased keywords are not allowed in STRICT mode
+    boolean allowsUpperCased = strictness != Strictness.STRICT;
+
+    // Confirm that chars [0..length) match the keyword.
     int length = keyword.length();
-    for (int i = 1; i < length; i++) {
+    for (int i = 0; i < length; i++) {
       if (pos + i >= limit && !fillBuffer(i + 1)) {
         return PEEKED_NONE;
       }
       c = buffer[pos + i];
-      if (c != keyword.charAt(i) && c != keywordUpper.charAt(i)) {
+      if (c != keyword.charAt(i) && !(allowsUpperCased && c == keywordUpper.charAt(i))) {
```

Comment:
The negation makes this condition a little hard to follow. What would you think of something like this?
```java
boolean matched = c == keyword.charAt(i) || (allowsUpperCased && c == keywordUpper.charAt(i));
if (!matched) {
  return PEEKED_NONE;
}
```